include/aws/http/connection.h

```diff
@@ -334,9 +393,15 @@ AWS_HTTP_API
 struct aws_channel *aws_http_connection_get_channel(struct aws_http_connection *connection);
 
 /**
- * HTTP/2 specific. Change the HTTP/2 conenction settings.
- * Settings frame will be sent. If set, on_completed callback will be always be invoked, if no error is reported then
- * the peer has acknowledged the settings and the change has been applied.
+ * Send a SETTINGS frame (HTTP/2 only).
+ * SETTINGS will be applied locally when SETTINGS ACK is received from peer.
+ *
+ * @param http2_connection HTTP/2 connection.
+ * @param settings_array The array of settings to change. Note: each setting has its boundary.
+ * @param num_settings The num of settings to change in settings_array.
+ *      If it's zero, we will just do nothing, no frame will be sent. And callback will be ignored.
```

Comment:
regarding num_settings=0: It's weird to have the function return AWS_OP_SUCCESS, but then the callback doesn't fire. 

Let's do one of these things:
1) Let's just send the empty SETTINGS frame. It's not against the law. That's what they asked us to do.
2) It's an error to call this with num_settings=0